### ISSUE TITLE
allow users to add todos

### ISSUE THREAD

**Issue by lavanyagarg112** [lavanyagarg112/meetwise#59]
/add-todo
POST

ill send:

the task
the deadline: date and time

Manual stuff:
assigner: 
assignee: 

Completed:
boolean

so ill send -

{
   meetingid:
   organisation:
   team: can be null
   details:
   deadline: datetime
   assigner: Person schema
   assignee: Person schema
   isCompleted: boolean
}

/edit-todo

PUT request
send all the details as add-todo but update request

/delete-todo
delete request

per meeting so ill send meeting id, org and team, todo id
{
   meetingid:
   organisation:
   team: can be null
   todoid
}

/get-meeting-todos
array of todos

to return:

todos: [ {}, {} ] where each object is:

{
   id: todo id
   details:
   deadline: datetime
   assigner: Person schema
   assignee: Person schema
   isCompleted: boolean
}

what ill send:
{
   meetingid:
   organisation:
   team: can be null
}

/get-user-todos
post request

basically get-meeting-todos but where user is the assignee or assigner

todos: { assigner: [ {}, {} ], assignee: [ {}, {} }
so assigner array is the list of all todos where user is the assinger
assignee array is the list of all todos where user is the assignee

what ill send:
{
   meetingid:
   organisation:
   team: can be null
}

/get-all-user-todos
get request

basically /get-user-todos but for all the meetings in the teams that the user can see from active organisation

**Comment by lavanyagarg112** [comment]
But @Timenikhil I have updated the schema. here is the same thing for your ref:

TODOS:

/add-todo
POST

what i will send:

{
    meetingid:
    organisation:
    team: can be null
    details:
    deadline: datetime
    assigner: Person schema
    assignee: Person schema
    isCompleted: boolean
}

what i want:

(im guessing this can be ur todo schema cause ill need all todos in this format)
{
    id: todo id
    details:
    deadline: datetime
    assigner: Person schema
    assignee: Person schema
    isCompleted: boolean
}

/edit-todo
PUT

send all the details in add-todo but update request


/delete-todo
DELETE

per meeting so ill send meeting id, org and team, todo id
{
    meetingid:
    organisation:
    team: can be null
    todoid:
}

/get-meeting-todos
POST

what i want:
{
    todos: [
        {
            id: todo id
            details:
            deadline: datetime
            assigner: Person schema
            assignee: Person schema
            isCompleted: boolean
        },
        {
            id: todo id
            details:
            deadline: datetime
            assigner: Person schema
            assignee: Person schema
            isCompleted: boolean
        }

    ]
}

what ill send:
{
    meetingid:
    organisation:
    team: can be null
}


/get-user-todos
POST

basically get-meeting-todos but where sent user in post request is the assignee or assigner

What I want:

{
    todos: {
        assigner: [
            {
                id: todo id
                details:
                deadline: datetime
                assigner: Person schema
                assignee: Person schema
                isCompleted: boolean
            },
            {
                id: todo id
                details:
                deadline: datetime
                assigner: Person schema
                assignee: Person schema
                isCompleted: boolean
            }
        ],

        assignee: [
            {
                id: todo id
                details:
                deadline: datetime
                assigner: Person schema
                assignee: Person schema
                isCompleted: boolean
            },
            {
                id: todo id
                details:
                deadline: datetime
                assigner: Person schema
                assignee: Person schema
                isCompleted: boolean
            }
        ]

    }
}

so assigner array is the list of all todos where user is the assinger
assignee array is the list of all todos where user is the assignee

what ill send:
{
    userId: userid
    meetingid:
    organisation:
    team: can be null
}


/get-all-user-todos
GET

basically /get-user-todos but for all the meetings in the teams that the CURRENT user can see from active organisation
so ill not send you anything. you can use current user id from credentials for this

**Comment by lavanyagarg112** [comment]
@Timenikhil you can see the updated schema in main
